lib/mongo/collection/view/readable.rb

```diff
@@ -384,7 +402,8 @@ def limit(limit = nil)
         #
         # @since 2.0.0
         def map_reduce(map, reduce, options = {})
-          MapReduce.new(self, map, reduce, @options.merge(options))
+          options = @options.slice(:limit, :sort).merge(options)
+          MapReduce.new(self, map, reduce, options)
```

Comment:
This also looks like a meaningful behavior change.

Reply:
See my comment on this above. What do you think about this change?